src/api-serverless/src/nextgen/nextgen.routes.ts

```diff
@@ -452,6 +452,69 @@ router.get(
   }
 );
 
+router.get(
+  `/collections/:id/ultimate_trait_set`,
+  async function (req: any, res: any, next: any) {
+    const id: number = parseInt(req.params.id);
+    const traits = req.query.trait;
+
+    if (isNaN(id)) {
+      return res.status(404).send({});
```

Comment:
Same thing. Throw a BadRequestException with explanatory reason instead. Same with the "traits" thing later on.